kratos/python_scripts/apply_periodic_boundaray_condition_process.py

```diff
@@ -0,0 +1,115 @@
+import KratosMultiphysics as KratosMultiphysics
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ApplyPeriodicBoundaryConditionProcess(Model, settings["Parameters"])
+
+## All the processes python should be derived from "Process"
+class ApplyPeriodicBoundaryConditionProcess(KratosMultiphysics.Process):
+    """This class is used in order to impose a rigid body movement in a certain region of the problem
+
+    This class constructs the model parts containing the constrains that enforce the rigid body movement
+    Only the member variables listed below should be accessed directly.
+
+    Public member variables:
+    Model -- the container of the different model parts.
+    settings -- Kratos parameters containing solver settings.
+    """
+
+    def __init__(self, Model, settings ):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        Model -- the container of the different model parts.
+        settings -- Kratos parameters containing solver settings.
+        """
+        KratosMultiphysics.Process.__init__(self)
+
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "help"                        : "This process uses LinearMasterSlaveConstraint in order to impose a periodic boundary condition on the given submodelparts. The process takes the first provided submodelpart as master and the second as slave.",
+            "computing_model_part_name"   : "",
+            "model_part_name"             : "please_specify_model_part_name",
+            "first_model_part_name"       : "please_specify_model_part_name",
+            "second_model_part_name"      : "please_specify_model_part_name",
+            "interval"                    : [0.0, 1e30],
+            "variable_names"              : [],
+            "center"                      : [0,0,0],
+            "axis_of_rotation"            : [0.0,0.0,0.0],
```

Comment:
We should avoid a parameters which includes two separate problems (rotation and translation) where, depending on the problem, only a subset of the parameters makes sense.

Reply:
The reason I made it into a combined file is that only thing that changes from the translation to rotation case are the matrices `mTransformationMatrix` and `mTransformationMatrixVariable` which are used to transform the geometry and the variable between master and slave modelparts and to form the constraints. 
So for me its easy to have this here in the class and not having same implementation (may be inheritance is an option here between rotational periodic and transnational periodic boundary condition) for two different processes.

Reply:
I really think this should be refactored. It should be relatively simple to do without inheritance. We can discuss it in person.